dapps/tests/app/test/simple_storage_spec.js

```diff
@@ -50,14 +50,23 @@ contract("SimpleStorage", function() {
     assert.strictEqual(SimpleStorage.options.address, SimpleStorage.address);
   });
 
-  it('listens to events', function(done) {
-    SimpleStorage.once('EventOnSet2', async function(error, result) {
-      assert.strictEqual(error, null);
-      assert.strictEqual(parseInt(result.returnValues.setValue, 10), 150);
-      done(error);
+  it('listens to events', async function() {
+    const promise = new Promise((resolve) => {
+      SimpleStorage.once("EventOnSet2", (error, result) => {
+        assert.strictEqual(error, null);
```

Comment:
It's probably a good idea, generally in setups like this, to wrap the asserts in try/catch and pass any error along with `reject`; note that `new Promise((resolve) => {` will need to become `new Promise((resolve, reject) => {`

```js
try {
  assert.strictEqual(error, null);
  assert.strictEqual(parseInt(result.returnValues.setValue, 10), 150);
} catch (err) {
  reject(err);
}    
resolve();
```